contracts/chain-adapters/ForwarderBase.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+/**
+ * @title ForwarderBase
+ * @notice Base contract deployed some network between L1 and the network housing a spoke pool contract. If a message is sent
+ * to this contract which came from the cross domain admin, then it is routed to the next network via the canonical messaging
+ * bridge (which may be done automatically by the network's finalization infrastructure). Tokens sent from L1 are stored temporarily
+ * on this contract. Any EOA can initiate a bridge of these tokens to the target `l3SpokePool`.
+ * @custom:security-contact bugs@across.to
+ */
+abstract contract ForwarderBase is UUPSUpgradeable {
+    // L3 address of the recipient of L1 messages and tokens.
+    address public l3SpokePool;
```

Comment:
We might want to note either here or in the rerouter_adapter that this l3 spoke pool address i only used in the contracts IN THIS CONTRACT, and its essentially useless in the `rerouter_adapter`. However, the developer/admin should still set the CrossChainContracts correctly including the correct L3 spoke pool address because the UMIP prescribes that this event contains the canonical mapping of chain Id's to spoke pool addresses

Reply:
I think it makes more sense to put that in the adapter contract: https://github.com/across-protocol/contracts/pull/607/commits/6cbbfd48618fff81f6027f9a88cc632757c1879c